src/Carbon/Carbon.php

```diff
@@ -772,11 +786,23 @@ public function tz($value)
      *
      * @param DateTimeZone|string $value
      *
+     * @throws Exception
+     *
      * @return static
      */
     public function setTimezone($value)
     {
-        return parent::setTimezone(static::safeCreateDateTimeZone($value));
+        try {
+            return parent::setTimezone(static::safeCreateDateTimeZone($value));
+        } catch (Exception $e) {
+            // @codeCoverageIgnoreStart
+            $message = $e->getMessage();
+            if (self::isHhvm() && 0 === strpos($message, 'DateTimeZone::__construct()')) {
```

Comment:
We don't use yoda conditions, `strpos($message, 'DateTimeZone::__construct()') === 0` is to be chosen